pkg/analysis/nodurations/analyzer.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodurations
+
+import (
+	"fmt"
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+	kalerrors "sigs.k8s.io/kube-api-linter/pkg/analysis/errors"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/utils"
+)
+
+const name = "nodurations"
+
+// Analyzer is the analyzer for the nodurations package.
+// It checks that no struct field is of a type either time.Duration or metav1.Duration.
+var Analyzer = &analysis.Analyzer{
+	Name:     name,
+	Doc:      "Use seconds instead of Duration to avoid need to implement go duration parsing",
+	Run:      run,
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+}
+
+func run(pass *analysis.Pass) (any, error) {
+	inspect, ok := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	if !ok {
+		return nil, kalerrors.ErrCouldNotGetInspector
+	}
+
+	// Filter to fields so that we can look at fields within structs.
+	// Filter typespecs so that we can look at type aliases.
+	nodeFilter := []ast.Node{
+		(*ast.StructType)(nil),
+		(*ast.TypeSpec)(nil),
+	}
+
+	inspect.Preorder(nodeFilter, func(node ast.Node) {
+		switch n := node.(type) {
+		case *ast.StructType:
+			if n.Fields == nil {
+				return
+			}
+
+			for _, field := range n.Fields.List {
+				checkField(pass, field)
+			}
+		case *ast.Field:
+			checkField(pass, n)
+		case *ast.TypeSpec:
+			checkTypeSpec(pass, n, node, "type")
+		}
+	})
+
+	return nil, nil //nolint:nilnil
+}
+
+func checkField(pass *analysis.Pass, field *ast.Field) {
+	fieldName := utils.FieldName(field)
+	if fieldName == "" {
+		return
+	}
+
+	prefix := fmt.Sprintf("field %s", fieldName)
+
+	checkTypeExpr(pass, field.Type, field, prefix)
+}
+
+//nolint:cyclop
+func checkTypeExpr(pass *analysis.Pass, typeExpr ast.Expr, node ast.Node, prefix string) {
+	switch typ := typeExpr.(type) {
+	case *ast.SelectorExpr:
+		pkg, ok := typ.X.(*ast.Ident)
+		if !ok {
+			return
+		}
+
+		if typ.X == nil || (pkg.Name != "time" && pkg.Name != "metav1") {
+			return
+		}
+
+		// Array element is not a metav1.Condition.
+		if typ.Sel == nil || typ.Sel.Name != "Duration" {
+			return
+		}
+
+		pass.Reportf(node.Pos(), "%s should not use a Duration. Use seconds to avoid go Duration parsing.", prefix)
```

Comment:
```suggestion
		pass.Reportf(node.Pos(), "%s should not use a Duration. Use an integer type with units in the name to avoid the need for clients to implement Go style duration parsing.", prefix)
```